clients/command-pubsub/src/test/java/org/eclipse/hono/client/command/pubsub/PubSubBasedCommandTest.java

```diff
@@ -150,13 +171,45 @@ public void testFromMessageFailsForMissingCorrelationIdWithResponseRequired() {
         assertThat(command.getInvalidCommandReason()).contains("correlation-id is not set");
     }
 
+    /**
+     * Verifies that an invalid command is created from a PubsubMessage that does not contain a correlation-id attribute
+     * but has ack-required set to true.
+     */
+    @Test
+    public void testFromMessageFailsForMissingCorrelationIdWithAckRequired() {
+        final String ackRequired = "true";
+        final PubsubMessage message = getPubSubMessage(tenantId, deviceId, subject, null, null, ackRequired,
+                null);
+
+        final PubSubBasedCommand command = PubSubBasedCommand.from(message, tenantId);
+
+        assertFalse(command.isValid());
+        assertThat(command.getInvalidCommandReason()).contains("correlation-id is not set");
```

Comment:
verifying string literals in error messages makes the test cases very prone to failing when the text is being slightly adjusted/altered. FMPOV we should not do such checks. If we need to verify specific causes, then we should introduce specific exceptions or codes instead ...

Reply:
I see your point, but that change would need to be made on the Command Interface level. So for now I just changed the checks to be inline with the ones for the Kafka- and ProtonBasedCommandTest.